diff -urN initscripts-9.05.orig/rc.d/rc.sysinit initscripts-9.05/rc.d/rc.sysinit
--- initscripts-9.05.orig/rc.d/rc.sysinit	2010-03-15 18:45:14.139172865 +0300
+++ initscripts-9.05/rc.d/rc.sysinit	2010-03-19 16:39:27.986051518 +0300
@@ -258,22 +258,9 @@
 # Print a text banner.
 echo -en $"\t\tWelcome to "
 read -r redhat_release < /etc/redhat-release
-if [[ "$redhat_release" == *"Red Hat"* ]]; then
- [ "$BOOTUP" = "color" ] && echo -en "\\033[0;31m"
- echo -en "Red Hat"
- [ "$BOOTUP" = "color" ] && echo -en "\\033[0;39m"
- PRODUCT=$(sed "s/Red Hat \(.*\) release.*/\1/" /etc/redhat-release)
- echo " $PRODUCT"
-elif [[ "$redhat_release" == *Fedora* ]]; then
  [ "$BOOTUP" = "color" ] && echo -en "\\033[0;34m"
- echo -en "Fedora"
+ echo -e "RFRemix"
  [ "$BOOTUP" = "color" ] && echo -en "\\033[0;39m"
- PRODUCT=$(sed "s/Fedora \(.*\) \?release.*/\1/" /etc/redhat-release)
- echo " $PRODUCT"
-else
- PRODUCT=$(sed "s/ release.*//g" /etc/redhat-release)
- echo "$PRODUCT"
-fi
 if [ "$PROMPT" != "no" ]; then
  echo -en $"\t\tPress 'I' to enter interactive startup."
  echo
