diff -ruHpN initscripts-8.76.2.orig/udev/rules.d/96-olpc-keyboard.rules initscripts-8.76.2/udev/rules.d/96-olpc-keyboard.rules
--- initscripts-8.76.2.orig/udev/rules.d/96-olpc-keyboard.rules	1969-12-31 19:00:00.000000000 -0500
+++ initscripts-8.76.2/udev/rules.d/96-olpc-keyboard.rules	2008-07-24 12:44:36.000000000 -0400
@@ -0,0 +1 @@
+ACTION=="add", SYSFS{name}=="AT Translated Set 2 keyboard", KERNEL=="event*", RUN+="/usr/sbin/setolpckeys /dev/input/%k"
